src/Raven.Server/Documents/ETL/Providers/Queue/AzureQueueStorage/AzureQueueStorageEtl.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Storage.Queues;
+using CloudNative.CloudEvents;
+using CloudNative.CloudEvents.Extensions;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Client.Documents.Operations.ETL.Queue;
+using Raven.Server.Documents.ETL.Stats;
+using Raven.Server.Exceptions.ETL.QueueEtl;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.ETL.Providers.Queue.AzureQueueStorage;
+
+public sealed class AzureQueueStorageEtl : QueueEtl<AzureQueueStorageItem>
+{
+    private readonly HashSet<string> _alreadyCreatedQueues = new();
+    private readonly Dictionary<string, QueueClient> _queueClients = new();
+
+    public AzureQueueStorageEtl(Transformation transformation, QueueEtlConfiguration configuration,
+        DocumentDatabase database, ServerStore serverStore) : base(transformation, configuration, database, serverStore)
+    {
+    }
+
+    protected override
+        EtlTransformer<QueueItem, QueueWithItems<AzureQueueStorageItem>, EtlStatsScope, EtlPerformanceOperation>
+        GetTransformer(DocumentsOperationContext context)
+    {
+        return new AzureQueueStorageDocumentTransformer<AzureQueueStorageItem>(Transformation, Database, context,
+            Configuration);
+    }
+
+    protected override int PublishMessages(List<QueueWithItems<AzureQueueStorageItem>> itemsPerQueue,
+        BlittableJsonEventBinaryFormatter formatter, out List<string> idsToDelete)
+    {
+        if (itemsPerQueue.Count == 0)
+        {
+            idsToDelete = null;
+            return 0;
+        }
+
+        idsToDelete = [];
+        int count = 0;
+
+        foreach (QueueWithItems<AzureQueueStorageItem> queue in itemsPerQueue)
+        {
+            string queueName = queue.Name.ToLower();
+
+            if (_queueClients.TryGetValue(queueName, out QueueClient queueClient) == false)
+            {
+                queueClient =
+                    QueueBrokerConnectionHelper.CreateAzureQueueStorageClient(
+                        Configuration.Connection.AzureQueueStorageConnectionSettings, queueName);
+                _queueClients.Add(queueName, queueClient);
+            }
+
+            if (Configuration.SkipAutomaticQueueDeclaration == false)
+                CreateQueueIfNotExists(queueClient);
+
+            foreach (AzureQueueStorageItem queueItem in queue.Items)
+            {
```

Comment:
I think we should have `CancellationToken.ThrowIfCancellationRequested();` here in order to stop processing if the cancellation was requested